### PR TITLE
Update documentation

### LINKED ISSUE #2106: Update documentation
## Documentation Request

For the 4.0 release, we should update the docs to reflect recent changes on Sonata.

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is the updated docs for 4.0.

Closes #2106

### CODE REVIEW

File: docs/reference/creating_a_provider_class.rst [hunk 1]
```diff
@@ -345,11 +348,11 @@ of options that need to be passed to the ``view_vimeo.html.twig`` template file:
...
         $params = [
-            'src'         => http_build_query($player_parameters),
-            'id'          => $player_parameters['js_swf_id'],
-            'frameborder' => isset($options['frameborder']) ? $options['frameborder'] : 0,
-            'width'       => isset($options['width']) ? $options['width'] : $media->getWidth(),
-            'height'      => isset($options['height']) ? $options['height'] : $media->getHeight(),
+            'src' => http_build_query($player_parameters),
+            'id' => $player_parameters['js_swf_id'],
+            'frameborder' => $options['frameborder'] ?? 0,
+            'width' => $options['width'] ?? : $media->getWidth(),
```
Comment (jordisala1991) [inline docs/reference/creating_a_provider_class.rst]: wrong ?? :

File: docs/reference/helpers.rst [hunk 2]
```diff
@@ -6,33 +6,6 @@ itself. The thumbnail always represents the media's image preview (i.e. the
 thumbnail of a flash video). And the media helper generates the media itself
 (i.e. the flash video).
 
-PHP Usage
```
Comment (jordisala1991) [inline docs/reference/helpers.rst:9]: Not sure if that works, but looks like a strange way of using the functions. @sonata-project/contributors

File: docs/reference/advanced_configuration.rst [hunk 3]
```diff
@@ -16,7 +16,7 @@ Full configuration options:
         force_disable_category: false # true, if you really want to disable the relation with category
 
         default_context: default
-        admin_format:   { width: 200 , quality: 90, format: 'jpg'}
+        admin_format: { width: 200 , quality: 90, format: 'jpg'}
```
Comment (VincentLanglet) [inline docs/reference/advanced_configuration.rst]:
```suggestion
        admin_format: {width: 200, quality: 90, format: 'jpg'}
```

File: docs/reference/advanced_configuration.rst [hunk 4]
```diff
@@ -30,9 +30,9 @@ Full configuration options:
 
                 formats:
                     small: { width: 100 , quality: 70}
-                    big:   { width: 500 , quality: 70, resizer: sonata.media.resizer.square}
+                    big: { width: 500 , quality: 70, resizer: sonata.media.resizer.square}
```
Comment (VincentLanglet) [inline docs/reference/advanced_configuration.rst]:
```suggestion
                    big: {width: 500, quality: 70, resizer: sonata.media.resizer.square}
```

Same for other, the spacing is weird

File: docs/reference/media_context.rst [hunk 5]
```diff
@@ -31,7 +31,7 @@ configured in the provider:
 
                 formats:
                     small: { width: 100 , quality: 70}
-                    big:   { width: 500 , quality: 70, resizer: sonata.media.resizer.square}
+                    big: { width: 500 , quality: 70, resizer: sonata.media.resizer.square}
```
Comment (VincentLanglet) [inline docs/reference/media_context.rst]:
```suggestion
                    big: {width: 500, quality: 70, resizer: sonata.media.resizer.square}
```

same for others

File: docs/reference/advanced_configuration.rst (hunk 3 above)
Comment (jordisala1991) [reply docs/reference/advanced_configuration.rst]: I decided to add a space between { and the width, I think it reads better that way.

File: docs/reference/advanced_configuration.rst (hunk 4 above)
Comment (jordisala1991) [reply docs/reference/advanced_configuration.rst]: Fixed all spacing

File: docs/reference/media_context.rst (hunk 5 above)
Comment (jordisala1991) [reply docs/reference/media_context.rst]: Fixed.

**Review by VincentLanglet — DISMISSED** [review]

**Comment by jordisala1991** [comment]
Please review @sonata-project/contributors

File: docs/index.rst [hunk 6]
```diff
@@ -44,24 +44,25 @@ Available services
...
-    - ``sonata.media.provider.image``         : Image
-    - ``sonata.media.provider.file``          : File
-    - ``sonata.media.provider.dailymotion``   : Dailymotion
-    - ``sonata.media.provider.vimeo``         : Vimeo
-    - ``sonata.media.provider.youtube``       : Youtube
+    - ``sonata.media.provider.image``: Image
+    - ``sonata.media.provider.file``: File
+    - ``sonata.media.provider.dailymotion``: Dailymotion
+    - ``sonata.media.provider.vimeo``: Vimeo
+    - ``sonata.media.provider.youtube``: Youtube
```
Comment (phansys) [inline docs/index.rst]:
```suggestion
    - ``sonata.media.provider.youtube``: [YouTube](https://www.youtube.com/)
```

File: docs/index.rst [hunk 7]
```diff
@@ -44,24 +44,25 @@ Available services
...
  -  Filesystem
 
-    - ``sonata.media.filesystem.local``       : The local filesystem (default)
-    - ``sonata.media.filesystem.ftp``         : FTP
-    - ``sonata.media.filesystem.s3``          : Amazon S3
-    - ``sonata.media.filesystem.replicate``   : Replicate file to a master and to a slave
+    - ``sonata.media.filesystem.local``: The local filesystem (default)
+    - ``sonata.media.filesystem.ftp``: FTP
+    - ``sonata.media.filesystem.s3``: Amazon S3
+    - ``sonata.media.filesystem.replicate``: Replicate file to a master and to a slave
```
Comment (phansys) [inline docs/index.rst:58]: I thought currently we were using "primary" and "secondary", but checking #1778 it seems the change was only partial.

File: docs/index.rst [hunk 8]
```diff
@@ -44,24 +44,25 @@ Available services
...
 
  - CDN
 
-    - ``sonata.media.cdn.server``             : The local http server (default)
-    - ``sonata.media.cdn.panther``            : Panther Portal
-    - ``sonata.media.cdn.fallback``           : Fallback, use the fallback (the http server) if the Media is not yet flushed on the CDN
+    - ``sonata.media.cdn.server``: The local http server (default)
```
Comment (phansys) [inline docs/index.rst]:
```suggestion
    - ``sonata.media.cdn.server``: The local HTTP server (default)
```

File: docs/index.rst [hunk 9]
```diff
@@ -44,24 +44,25 @@ Available services
...
+    - ``sonata.media.cdn.panther``: Panther Portal
+    - ``sonata.media.cdn.cloudfront``: Amazon Cloudfront CDN
```
Comment (phansys) [inline docs/index.rst]:
```suggestion
    - ``sonata.media.cdn.cloudfront``: [AWS CloudFront](https://aws.amazon.com/cloudfront/)
```

File: docs/index.rst [hunk 10]
```diff
@@ -44,24 +44,25 @@ Available services
...
 
-    - ``sonata.media.provider.image``         : Image
-    - ``sonata.media.provider.file``          : File
-    - ``sonata.media.provider.dailymotion``   : Dailymotion
-    - ``sonata.media.provider.vimeo``         : Vimeo
-    - ``sonata.media.provider.youtube``       : Youtube
+    - ``sonata.media.provider.image``: Image
+    - ``sonata.media.provider.file``: File
+    - ``sonata.media.provider.dailymotion``: Dailymotion
+    - ``sonata.media.provider.vimeo``: Vimeo
```
Comment (phansys) [inline docs/index.rst]:
```suggestion
    - ``sonata.media.provider.vimeo``: [Vimeo](https://vimeo.com/)
```

File: docs/index.rst [hunk 11]
```diff
@@ -44,24 +44,25 @@ Available services
...
  - Providers
 
-    - ``sonata.media.provider.image``         : Image
-    - ``sonata.media.provider.file``          : File
-    - ``sonata.media.provider.dailymotion``   : Dailymotion
-    - ``sonata.media.provider.vimeo``         : Vimeo
-    - ``sonata.media.provider.youtube``       : Youtube
+    - ``sonata.media.provider.image``: Image
+    - ``sonata.media.provider.file``: File
+    - ``sonata.media.provider.dailymotion``: Dailymotion
```
Comment (phansys) [inline docs/index.rst]:
```suggestion
    - ``sonata.media.provider.dailymotion``: [Dailymotion](https://www.dailymotion.com/)
```

File: docs/reference/advanced_configuration.rst [hunk 12]
```diff
@@ -58,116 +58,106 @@ Full configuration options:
...
             panther:
-                path:       http://domain.pantherportal.com/uploads/media
+                path: http://domain.pantherportal.com/uploads/media
                 site_id:
                 password:
                 username:
 
             cloudfront:
-                path:       http://xxxxxxxxxxxxxx.cloudfront.net/uploads/media
+                path: http://xxxxxxxxxxxxxx.cloudfront.net/uploads/media
```
Comment (phansys) [inline docs/reference/advanced_configuration.rst]:
```suggestion
                path: https://xxxxxxxxxxxxxx.cloudfront.net/uploads/media
```

File: docs/reference/advanced_configuration.rst [hunk 13]
```diff
@@ -58,116 +58,106 @@ Full configuration options:
...
+                    big: { width: 500, quality: 90 }
 
         cdn:
             server:
-                path:      /uploads/media # http://media.sonata-project.org
+                path: /uploads/media # http://media.sonata-project.org
 
             panther:
-                path:       http://domain.pantherportal.com/uploads/media
+                path: http://domain.pantherportal.com/uploads/media
```
Comment (phansys) [inline docs/reference/advanced_configuration.rst:69]: I think this provider does not exist anymore.

File: docs/reference/creating_a_provider_class.rst [hunk 14]
```diff
@@ -18,7 +18,7 @@ asset:
 
 A provider class is always linked to a ``Filesystem`` and a ``CDN``. The
 filesystem abstraction uses the ``Gaufrette`` library. For now there is
-only 2 abstracted filesystem available : ``Local`` and ``FTP``. The ``CDN``
+only 2 abstracted filesystem available: ``Local`` and ``FTP``. The ``CDN``
 is used to generated the media asset public url.
```
Comment (phansys) [inline docs/reference/creating_a_provider_class.rst]:
```suggestion
is used to generated the media asset public URL.
```

File: docs/reference/creating_a_provider_class.rst [hunk 15]
```diff
@@ -48,31 +48,31 @@ Case Study
 Before starting, we need to collect information about some asset on vimeo.
 Take this video, for example:
 
-* video identifier format : 21216091
-* video player documentation : http://vimeo.com/api/docs/moogaloop
-* metadata : https://vimeo.com/api/oembed.json?url=http://vimeo.com/21216091
+* video identifier format: 21216091
+* video player documentation: http://vimeo.com/api/docs/moogaloop
```
Comment (phansys) [inline docs/reference/creating_a_provider_class.rst]:
```suggestion
* video player documentation: https://developer.vimeo.com/player/sdk
```

File: docs/reference/creating_a_provider_class.rst [hunk 16]
```diff
@@ -48,31 +48,31 @@ Case Study
...
+* metadata: https://vimeo.com/api/oembed.json?url=http://vimeo.com/21216091
```
Comment (phansys) [inline docs/reference/creating_a_provider_class.rst]:
```suggestion
```

File: docs/reference/helpers.rst [hunk 17]
```diff
@@ -102,8 +75,11 @@ Thumbnails for files
 --------------------
 
 The ``sonata.media.provider.file`` provider does not generate thumbnails.
-This provider tries to display a default thumbnail by context and format.
+This provider tries to display a default thumbnail.
+
+The default thumbnail must be put in the ``bundles/sonatamedia/file.png``.
```
Comment (phansys) [inline docs/reference/helpers.rst]:
```suggestion
The default thumbnail must be put in ``bundles/sonatamedia/file.png``.
```

File: docs/reference/helpers.rst [hunk 18]
```diff
@@ -102,8 +75,11 @@ Thumbnails for files
...
+It is automatically there when you install assets using
```
Comment (phansys) [inline docs/reference/helpers.rst]:
```suggestion
It is automatically available there when you install assets using
```

File: docs/reference/media_context.rst [hunk 19]
```diff
@@ -60,32 +60,22 @@ Doctrine ORM:
...
 
-    public function configureFormFields(FormMapper $form)
+    public function configureFormFields(FormMapper $form): void
     {
         $form
             ->add('image', 'sonata_type_model_list', [], ['link_parameters' =>['context' => 'news']]);
     }
 
-If you look in the ``MediaAdmin`` class, the class defined a ``getPersistentParameters``
+If you look in the ``MediaAdmin`` class, the class defined a ``configurePersistentParameters``
```
Comment (phansys) [inline docs/reference/media_context.rst]:
```suggestion
If you look in the ``MediaAdmin`` class, the class defined a ``configurePersistentParameters()``
```

File: docs/reference/security.rst [hunk 20]
```diff
@@ -7,19 +7,19 @@ a download strategy interface, which can be set per context and authorize the me
...
 
-* ``sonata.media.security.superadmin_strategy`` : DEFAULT - the user needs to have one of the following roles :
+* ``sonata.media.security.superadmin_strategy``: DEFAULT - the user needs to have one of the following roles:
   ``ROLE_SUPER_ADMIN`` or ``ROLE_ADMIN`` (although these roles can be configured in SonataAdminBundle)
-* ``sonata.media.security.public_strategy`` : no restriction, files are public
-* ``sonata.media.security.forbidden_strategy`` : not possible to retrieve the original file
-* ``sonata.media.security.connected_strategy`` : the need to have one of the following roles :
+* ``sonata.media.security.public_strategy``: no restriction, files are public
+* ``sonata.media.security.forbidden_strategy``: not possible to retrieve the original file
+* ``sonata.media.security.connected_strategy``: the need to have one of the following roles:
```
Comment (phansys) [inline docs/reference/security.rst]:
```suggestion
* ``sonata.media.security.connected_strategy``: the user needs to have one of the following roles:
```

File: docs/reference/security.rst [hunk 21]
```diff
@@ -7,19 +7,19 @@ a download strategy interface, which can be set per context and authorize the me
...
   ``IS_AUTHENTICATED_FULLY`` or ``IS_AUTHENTICATED_REMEMBERED``
 
 On top of that, there is 3 download modes which can be configured to download the media. The download mode depends on
```
Comment (phansys) [inline docs/reference/security.rst]:
```suggestion
On top of that, there are 3 download modes which can be configured to download the media. The download mode depends on
```

File: docs/reference/security.rst [hunk 22]
```diff
@@ -62,19 +62,19 @@ The related download route name is ``sonata_media_download``.
 Creating your own Security Download Strategy
 --------------------------------------------
 
-The Strategy class must implement the ``DownloadStrategyInterface`` which contains 2 main methods :
+The Strategy class must implement the ``DownloadStrategyInterface`` which contains 2 main methods:
```
Comment (phansys) [inline docs/reference/security.rst]:
```suggestion
The strategy class must implement the ``DownloadStrategyInterface`` which contains 2 main methods:
```

File: docs/reference/security.rst [hunk 23]
```diff
@@ -115,10 +115,10 @@ Let's create the following strategy : a media can be downloaded only by the give
         }
     }
 
-Let's explain a bit :
+Let's explain a bit:
 
-* ``isGranted`` : the method test if granted user exists in allowed users for download
-* ``getDescription`` : return a translated message to explain what the current strategy does
+* ``isGranted``: the method test if granted user exists in allowed users for download
```
Comment (phansys) [inline docs/reference/security.rst]:
```suggestion
* ``isGranted``: the method test if allowed users contains the authenticated user for download
```

File: docs/reference/usage.rst [hunk 24]
```diff
@@ -30,7 +30,7 @@ or this::
 Retrieving metadata information
 -------------------------------
 
-The providers has a dedicated field where extra information can be saved : the providerMetadata field.
+The providers has a dedicated field where extra information can be saved: the providerMetadata field.
```
Comment (phansys) [inline docs/reference/usage.rst]:
```suggestion
The providers has a dedicated field where extra information can be saved: the ``providerMetadata`` field.
```

File: docs/reference/security.rst [hunk 25]
```diff
@@ -7,19 +7,19 @@ a download strategy interface, which can be set per context and authorize the me
...
 
 On top of that, there is 3 download modes which can be configured to download the media. The download mode depends on
 the HTTP server you used:
 
-* http : DEFAULT - use php to send the file
-* X-Sendfile : use the ``X-Sendfile`` flag (Apache + mod_xsendfile : https://tn123.org/mod_xsendfile/)
-* X-Accel-Redirect : use the ``X-Accel-Redirect`` flag (Nginx : http://wiki.nginx.org/X-accel)
+* http: DEFAULT - use php to send the file
+* X-Sendfile: use the ``X-Sendfile`` flag (Apache + mod_xsendfile: https://tn123.org/mod_xsendfile/)
+* X-Accel-Redirect: use the ``X-Accel-Redirect`` flag (Nginx: http://wiki.nginx.org/X-accel)
```
Comment (phansys) [inline docs/reference/security.rst]:
```suggestion
* X-Accel-Redirect: use the ``X-Accel-Redirect`` flag (Nginx: https://www.nginx.com/resources/wiki/start/topics/examples/x-accel/#x-accel-redirect)
```

**Review by phansys — CHANGES_REQUESTED** [review]

File: docs/index.rst (hunk 7 above)
Comment (jordisala1991) [reply docs/index.rst:58]: I will take a look at how is that implemented and if something is missing will open another PR.

File: docs/reference/advanced_configuration.rst (hunk 13 above)
Comment (jordisala1991) [reply docs/reference/advanced_configuration.rst:69]: Maybe it needs to be removed but it currently exists: https://github.com/sonata-project/SonataMediaBundle/blob/4.x/src/DependencyInjection/Configuration.php#L156-L166

Will take a look. Do you know why it should be removed, or where it was deprecated?

Comment (jordisala1991) [reply docs/reference/advanced_configuration.rst:69]: Anyway this will go on another PR because it currently exists

**Review by VincentLanglet — DISMISSED** [review]

File: docs/index.rst [hunk 26]
```diff
@@ -44,24 +44,25 @@ Available services
...
 
  - CDN
 
-    - ``sonata.media.cdn.server``             : The local http server (default)
-    - ``sonata.media.cdn.panther``            : Panther Portal
-    - ``sonata.media.cdn.fallback``           : Fallback, use the fallback (the http server) if the Media is not yet flushed on the CDN
+    - ``sonata.media.cdn.server``: The local HTTP server (default)
+    - ``sonata.media.cdn.panther``: Panther Portal
+    - ``sonata.media.cdn.cloudfront``: [AWS CloudFront](https://aws.amazon.com/cloudfront/)
+    - ``sonata.media.cdn.fallback``: Fallback, use the fallback (the http server) if the Media is not yet flushed on the CDN
```
Comment (phansys) [inline docs/index.rst]:
```suggestion
    - ``sonata.media.cdn.fallback``: Fallback, use the fallback (the HTTP server) if the media is not yet flushed on the CDN
```

File: docs/reference/advanced_configuration.rst (hunk 13 above)
Comment (phansys) [reply docs/reference/advanced_configuration.rst:69]: I'm not referring to our implementation, but the CDN provider itself.
BTW, I agree this must be resolved in a different PR.

File: docs/reference/creating_a_provider_class.rst [hunk 27]
```diff
@@ -345,11 +347,11 @@ of options that need to be passed to the ``view_vimeo.html.twig`` template file:
         $player_parameters =  array_merge($defaults, isset($options['player_parameters']) ? $options['player_parameters'] : []);
```
Comment (phansys) [inline docs/reference/creating_a_provider_class.rst]:
```suggestion
        $playerParameters =  array_merge($defaults, isset($options['player_parameters']) ? $options['player_parameters'] : []);
```

File: docs/reference/creating_a_provider_class.rst [hunk 28]
```diff
@@ -345,11 +347,11 @@ of options that need to be passed to the ``view_vimeo.html.twig`` template file:
...
 
         $params = [
-            'src'         => http_build_query($player_parameters),
-            'id'          => $player_parameters['js_swf_id'],
-            'frameborder' => isset($options['frameborder']) ? $options['frameborder'] : 0,
-            'width'       => isset($options['width']) ? $options['width'] : $media->getWidth(),
-            'height'      => isset($options['height']) ? $options['height'] : $media->getHeight(),
+            'src' => http_build_query($player_parameters),
```
Comment (phansys) [inline docs/reference/creating_a_provider_class.rst]:
```suggestion
            'src' => http_build_query($playerParameters),
```

File: docs/reference/creating_a_provider_class.rst [hunk 29]
```diff
@@ -345,11 +347,11 @@ of options that need to be passed to the ``view_vimeo.html.twig`` template file:
...
+            'id' => $player_parameters['js_swf_id'],
```
Comment (phansys) [inline docs/reference/creating_a_provider_class.rst]:
```suggestion
            'id' => $playerParameters['js_swf_id'],
```

**Review by phansys — CHANGES_REQUESTED** [review]

**Review by VincentLanglet — DISMISSED** [review]

**Comment by jordisala1991** [comment]
After this PR is merged I need to handle:

* Symfony 6 support (waiting for doctrine orm)
* Panther thing, Not sure what to do but something needs to be done
* Replicate thing, search for something to replace

**Review by phansys — APPROVED** [review]

File: docs/reference/troubleshooting.rst [hunk 30]
```diff
@@ -98,15 +93,15 @@ Finally your settings in your ``sonata_media`` parameters will look like this:
...
-                    small: { width: 100 , quality: 70  }
-                    big:   { width: 500 , quality: 70  }
-                    hq:    { width: 1920, quality: 100 }
+                    small: { width: 100, quality: 70 }
+                    big: { width: 500, quality: 70 }
+                    hq: { width: 1920, quality: 100 }
 
         cdn:
             server:
                 path: /uploads/media # http://media.sonata-project.org/
```
Comment (phansys) [inline docs/reference/troubleshooting.rst]:
```suggestion
                path: /uploads/media
```

File: docs/reference/advanced_configuration.rst [hunk 31]
```diff
@@ -58,116 +58,106 @@ Full configuration options:
...
                 formats:
-                    small: { width: 150 , quality: 95}
-                    big:   { width: 500 , quality: 90}
+                    small: { width: 150, quality: 95 }
+                    big: { width: 500, quality: 90 }
 
         cdn:
             server:
-                path:      /uploads/media # http://media.sonata-project.org
+                path: /uploads/media # http://media.sonata-project.org
```
Comment (phansys) [inline docs/reference/advanced_configuration.rst]:
```suggestion
                path: /uploads/media
```

**Review by phansys — DISMISSED** [review]

**Review by phansys — APPROVED** [review]

**Pull request merged**